internal/postgres/statement.go

```diff
@@ -0,0 +1,697 @@
+		return ddl.Date{}, nil
+	case "float8":
+		maxExpectedMods(0)
+		return ddl.Float64{}, nil
+	case "float4":
+		maxExpectedMods(0)
+		return ddl.Float64{}, []schemaIssue{widened}
+	case "int8":
+		maxExpectedMods(0)
+		return ddl.Int64{}, nil
+	case "int4":
+		maxExpectedMods(0)
+		return ddl.Int64{}, []schemaIssue{widened}
+	case "int2":
+		maxExpectedMods(0)
+		return ddl.Int64{}, []schemaIssue{widened}
+	case "numeric": // Map all numeric types to float64.
+		maxExpectedMods(2)
+		if len(mods) > 0 && mods[0] <= 15 {
+			// float64 can represent this numeric type faithfully.
+			// Note: int64 has 53 bits for mantissa, which is ~15.96
+			// decimal digits.
+			return ddl.Float64{}, []schemaIssue{numericThatFits}
+		}
+		return ddl.Float64{}, []schemaIssue{numeric}
+	case "serial":
+		maxExpectedMods(0)
+		return ddl.Int64{}, []schemaIssue{serial}
+	case "text":
+		maxExpectedMods(0)
+		return ddl.String{ddl.MaxLength{}}, nil
+	case "timestamptz":
+		maxExpectedMods(1)
+		return ddl.Timestamp{}, nil
+	case "timestamp":
+		maxExpectedMods(1)
+		// Map timestamp without timezone to Spanner timestamp.
+		return ddl.Timestamp{}, []schemaIssue{timestamp}
+	case "varchar":
+		maxExpectedMods(1)
+		if len(mods) > 0 {
+			return ddl.String{ddl.Int64Length{mods[0]}}, nil
+		}
+		return ddl.String{ddl.MaxLength{}}, nil
+	}
+	return ddl.String{ddl.MaxLength{}}, []schemaIssue{noGoodType}
+}
+
+func getTypeMods(conv *Conv, t nodes.List) (l []int64) {
+	for _, x := range t.Items {
+		switch t1 := x.(type) {
+		case nodes.A_Const:
+			switch t2 := t1.Val.(type) {
+			case nodes.Integer:
+				l = append(l, t2.Ival)
+			default:
+				conv.unexpected(fmt.Sprintf("Found %s node while processing Typmods", prNodeType(t1.Val)))
+			}
+		default:
+			conv.unexpected(fmt.Sprintf("Found %s node while processing Typmods", prNodeType(x)))
+		}
+	}
+	return l
+}
+
+func getArrayBounds(conv *Conv, t nodes.List) (l []int64) {
+	for _, x := range t.Items {
+		switch t := x.(type) {
+		case nodes.Integer:
+			// 'Ival' provides the array bound (-1 for an array where bound is not specified).
+			l = append(l, t.Ival)
+		default:
+			conv.unexpected(fmt.Sprintf("Found %s node while processing ArrayBounds", prNodeType(x)))
+		}
+	}
+	return l
+}
+
+func getTypeID(l []nodes.Node) (string, error) {
+	// The pg_query library generates a pg_catalog schema prefix for most
+	// types, but not for all. Typically "aliases" don't have the prefix.
+	// For example, "boolean" is parsed to ["pg_catalog", "bool"], but "bool" is
+	// parsed to ["bool"]. However the exact rules are unclear e.g. "date"
+	// is parsed to just ["date"].
+	// For simplicity, we strip off the pg_catalog prefix.
+	var ids []string
+	for _, t := range l {
+		s, err := getString(t)
+		if err != nil {
+			return "", err
+		}
+		ids = append(ids, s)
+	}
+	if len(ids) > 1 && ids[0] == "pg_catalog" {
+		ids = ids[1:]
+	}
+	return strings.Join(ids, "."), nil
+}
+
+func prType(pgCol pgColDef) string {
```

Comment:
Change to printType. I thought this was a typo for pgType.

Reply:
Done.